torch/nn/_functions/rnn.py

```diff
@@ -7,19 +7,22 @@
     pass
 
 
-def RNNReLUCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
-    hy = F.relu(F.linear(input, w_ih, b_ih) + F.linear(hidden, w_hh, b_hh))
+def RNNReLUCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None, skip_input=False):
+    xw_ih = input if skip_input else F.linear(input, w_ih, b_ih)
+    hy = F.relu(xw_ih + F.linear(hidden, w_hh, b_hh))
     return hy
 
 
-def RNNTanhCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
-    hy = F.tanh(F.linear(input, w_ih, b_ih) + F.linear(hidden, w_hh, b_hh))
+def RNNTanhCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None, skip_input=False):
+    xw_ih = input if skip_input else F.linear(input, w_ih, b_ih)
+    hy = F.tanh(xw_ih + F.linear(hidden, w_hh, b_hh))
     return hy
 
 
-def LSTMCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
+def LSTMCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None, skip_input=False):
     hx, cx = hidden
-    gates = F.linear(input, w_ih, b_ih) + F.linear(hx, w_hh, b_hh)
+    xw_ih = input.repeat(1, 4) if skip_input else F.linear(input, w_ih, b_ih)
```

Comment:
I've opened a branch [here](https://github.com/SeanNaren/pytorch/tree/skip_test) that you can build to see the issue. Running [this](https://gist.github.com/SeanNaren/1d5c636ab739a18516323214a226aea5) script. If there is a better way for you to replicate this problem let me know!